ooni/backend_client.py

```diff
@@ -0,0 +1,369 @@
+
+    def getInputPolicy(self):
+        return self.queryBackend('GET', '/policy/input')
+
+    def getNettestPolicy(self):
+        return self.queryBackend('GET', '/policy/nettest')
+
+    def getDeckList(self):
+        return self.queryBackend('GET', '/deck')
+
+    def getDeck(self, deck_hash):
+        from ooni.deck import Deck
+
+        deck = Deck(deck_hash)
+        if deck.descriptorCached:
+            return defer.succeed(deck)
+        else:
+            d = self.queryBackend('GET', '/deck/' + deck_hash)
+
+            @d.addCallback
+            def cb(descriptor):
+                deck.load(descriptor)
+                deck.save()
+                return deck
+
+            @d.addErrback
+            def err(err):
+                log.err("Failed to get descriptor for deck %s" % deck_hash)
+                log.exception(err)
+
+            return d
+
+    def downloadDeck(self, deck_hash):
+        from ooni.deck import Deck
+
+        deck = Deck(deck_hash)
+        if deck.fileCached:
+            return defer.succeed(deck)
+        else:
+            d = self.download('/deck/' + deck_hash + '/file', deck.cached_file)
+
+            @d.addCallback
+            def cb(res):
+                deck.verify()
+                return deck
+
+            @d.addErrback
+            def err(err):
+                log.err("Failed to download the deck %s" % deck_hash)
+                log.exception(err)
+
+            return d
+
+    def createReport(self, test_details):
+        request = {
+            'software_name': test_details['software_name'],
+            'software_version': test_details['software_version'],
+            'probe_asn': test_details['probe_asn'],
+            'probe_cc': test_details['probe_cc'],
+            'test_name': test_details['test_name'],
+            'test_version': test_details['test_version'],
+            'test_start_time': test_details['test_start_time'],
+            'input_hashes': test_details['input_hashes'],
+            'data_format_version': test_details['data_format_version'],
+            'format': 'json'
+        }
+        # import values from the environment
+        request.update([(k.lower(),v) for (k,v) in os.environ.iteritems()
+                        if k.startswith('PROBE_')])
+
+        return self.queryBackend('POST', '/report', query=request)
+
+    def updateReport(self, report_id, serialization_format, entry_content):
+        request = {
+            'format': serialization_format,
+            'content': entry_content
+        }
+        return self.queryBackend('POST', '/report/%s' % report_id,
+                                 query=request)
+
+
+    def closeReport(self, report_id):
+        return self.queryBackend('POST', '/report/' + report_id + '/close')
+
+class WebConnectivityClient(OONIBClient):
+    def isReachable(self):
+        # XXX maybe in the future we can have a dedicated API endpoint to
+        # test the reachability of the collector.
```

Comment:
I think this command does not apply to the web connectivity client, which has `/status`, right?